src/java/org/apache/cassandra/index/Index.java

```diff
@@ -136,6 +135,7 @@
  */
 public interface Index
 {
+    public static enum Loads {READS, WRITES, ALL, NONE};
```

Comment:
Using a plural name for a enum feels a little bit odd to me, what about something like `LoadType` or the more operation `OperationType`? Also I can't think of other future values for the enum given it's current usage, so I'm wondering if we could replace it by two booleans, implicit through a pair of `supportsWrites`/`supportsReads` methods. WDYT?

Reply:
I think it's a personal style thing. Rather than having 2 booleans and having to reason about them, the Enum makes it more 'in your face' imo: `ALL` is clear what it means vs 2 booleans etc. :shrug:

Also maybe I have too much imagination, but I can think of 'DELETE' as a load. You could have 'append-only' indexes in the future that don't support deletes. Maybe premature optimization is being my enemy here... Or sthg with statics for custom indexes...

I prefer you decide what is best here :-). I'll rename to singulars meanwhile in case we kept that option.